frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/EditCamperCard.tsx

```diff
@@ -0,0 +1,409 @@
+                    <Text
+                      textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}
+                    >
+                      Allergies
+                    </Text>
+                    <Text
+                      textStyle={{ sm: "xSmallRegular", lg: "buttonRegular" }}
+                    >
+                      Please list any allergies the camper has{" "}
+                    </Text>
+                  </FormLabel>
+
+                  <Textarea
+                    backgroundColor="#FFFFFF"
+                    value={camper.allergies ? camper.allergies : ""}
+                    onChange={(event) =>
+                      dispatchPersonalInfoAction({
+                        type: PersonalInfoActions.UPDATE_CAMPER,
+                        field: "allergies",
+                        camperIndex,
+                        data: event.target.value,
+                      })
+                    }
+                  />
+                </FormControl>
+              </WrapItem>
+
+              <Spacer />
+
+              <WrapItem width={{ sm: "100%", md: "47%" }}>
+                <FormControl>
+                  <FormLabel>
+                    <Text
+                      textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}
+                    >
+                      Special Needs
+                    </Text>
+                    <Text
+                      textStyle={{ sm: "xSmallRegular", lg: "buttonRegular" }}
+                    >
+                      Please list any special needs the camper has{" "}
+                    </Text>
+                  </FormLabel>
+
+                  <Textarea
+                    backgroundColor="#FFFFFF"
+                    value={camper.specialNeeds ? camper.specialNeeds : ""}
+                    onChange={(event) =>
+                      dispatchPersonalInfoAction({
+                        type: PersonalInfoActions.UPDATE_CAMPER,
+                        field: "specialNeeds",
+                        camperIndex,
+                        data: event.target.value,
+                      })
+                    }
+                  />
+                </FormControl>
+              </WrapItem>
+            </Wrap>
+          </Box>
+
+          <Box rounded="xl">
+            <Wrap>
+              <Spacer />
+              <WrapItem>
+                <Heading textStyle="displayLarge">
+                  <Flex py={6} px={{ sm: "5", lg: "20" }} alignItems="center">
+                    <Spacer />
+                    <Button
+                      variant="primary"
+                      textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}
+                      w={{ sm: "80px", lg: "200px" }}
+                      mr={4}
+                      h={{ sm: "30px", lg: "40px" }}
+                      onClick={updateFormErrorMsgs}
+                    >
+                      Save
+                    </Button>
+                    {CancelChangesModal()}
```

Comment:
From what I understand, this function returns the cancel button which also includes the deleteModal for discarding changes. Why could we not define that here? 

We could make this bottom part it's own component called  `EditCamperCardFooter` so the code is broken down a bit?

Reply:
Good idea, I've made a new component `EditCardFooter` that is a component for the bottom part, and this is where the logic for the cancellation modal is now,